# Fix: remove force_all_reduce param from finalize_model_grads
#
# Problem:
#   finalize_model_grads calls _BaseDataParallel.finish_grad_sync with force_all_reduce parameter
#   although it is only supported by the DistributedDataParallel class.
#   See https://github.com/NVIDIA/Megatron-LM/issues/4669
#
# Fix:
#   We add the parameter to the base class method signature.
#
# File: data_parallel_base.py, method _BaseDataParllel.finish_grad_sync, ~line 49.

diff --git a/megatron/core/distributed/data_parallel_base.py b/megatron/core/distributed/data_parallel_base.py
index b53152820..501484cb4 100644
--- a/megatron/core/distributed/data_parallel_base.py
+++ b/megatron/core/distributed/data_parallel_base.py
@@ -46,7 +46,7 @@ class _BaseDataParallel(MegatronModule):
         """Scale all gradients inside the buffers by `scaling_factor`."""
         pass
 
-    def finish_grad_sync(self):
+    def finish_grad_sync(self, force_all_reduce: bool = False):
         """
         Finishes grad sync (all-reduce or reduce-scatter) communication operations
         for all model gradients.
